catalyst/dl/experiment/config.py

```diff
@@ -537,5 +538,23 @@ def get_callbacks(self, stage: str) -> "OrderedDict[Callback]":
 
         return callbacks
 
+    def get_start_stage(self):
+        """
+        Return start stage, which may differ from `stages[0]` resuming
+        from checkpoint
+        """
+        state_params = self.get_state_params(
+            self.stages[0]
+        )
+        resume, resume_dir = [state_params.get(key, None)
+                              for key in ["resume", "resume_dir"]]
+
+        if resume_dir is not None:
+            resume = resume_dir / str(resume)
+
+        if resume is not None and Path(resume).is_file():
+            checkpoint = utils.load_checkpoint(resume)
+            return checkpoint["stage"]
+        return self.stages[0]
 
```

Comment:
this hack will not work, because we need such logic in every Experiment (not only Config one)

could you please change the logic for stages property for ConfigExperiment?
github.com/catalyst-team/catalyst/blob/master/catalyst/dl/experiment/config.py#L116#L119
this looks like a more elegant solution, with any changes in Runner logic

Reply:
@AndreySheka

Reply:
@Scitator I moved get_start_stage logic into stages property and rebased this branch to master. It's ready for merge.